status/settings.nim

```diff
@@ -92,13 +92,9 @@ proc getSettings*(db: DbConn): Settings =
     raise newException(ValueError, "No record found for settings")
   settings.get
 
-proc saveSetting*(db: DbConn, setting: string, value: auto)
-  {.deprecated("Please use saveSetting(SettingsCol) instead").} =
+proc saveSetting*(db: DbConn, setting: string | SettingsCol, value: auto) =
```

Comment:
Cool, i didn't know Nim supports alternative types for proc args! Can you please send a link to the Nim manual where it's described? I tried finding it myself but failed.

Reply:
I think it's allowed because `SettingsCol` type is a pure enum where all the enum values correspond to type `string`.

In which case, is there a reason to prefer `string | SettingsCol` to `SettingsCol`?

Reply:
> Cool, i didn't know Nim supports alternative types for proc args! Can you please send a link to the Nim manual where it's described? I tried finding it myself but failed.

The closest explanation in the manual I could find is in the [Generics Type Classes](https://nim-lang.org/docs/manual.html#generics-type-classes) and the [Implicit generics](https://nim-lang.org/docs/manual.html#generics-implicit-generics) sections:
> "Whilst the syntax of type classes appears to resemble that of ADTs/algebraic data types in ML-like languages, it should be understood that type classes are static constraints to be enforced at type instantiations. Type classes are not really types in themselves but are instead a system of providing generic "checks" that ultimately resolve to some singular type. Type classes do not allow for runtime type dynamism, unlike object variants or methods.
> A type class can be used directly as the parameter's type. Procedures utilizing type classes in such a manner are considered to be implicitly generic. They will be instantiated once for each unique combination of param types used within the program.
> By default, during overload resolution, each named type class will bind to exactly one concrete type. We call such type classes bind once types."

Reply:
> In which case, is there a reason to prefer `string | SettingsCol` over `SettingsCol`?

There was originally a deprecated proc `saveSetting` that accepted `setting: string`. I later added the second `saveSetting` proc that used `setting: SettingsCol` which called the original deprecated proc. Considering the deprecated proc was still called, it was still causing a compiler warning to be issued, and since the implementation of the deprecated proc simply used a `$` operation (via `fmt"{setting}"`), changing the parameter type to accept both `string` and `SettingCol` worked without changing the implementation, got rid of the deprecation, and still supported `string` as an alternative parameter type. Retaining support for `string` param type assumes that there are some external consumers of `nim-status` that are using this proc, however I highly doubt there are any in the wild, so **I've now changed this to simply remove `string` support, and only support `SettingsCol`**.